### PR TITLE
Add colorbar support to dataset

### PR DESCRIPTION
**Pull request by jenshnielsen**
@WilliamHPNielsen @jana-d

### CODE REVIEW

**Comment by WilliamHPNielsen** [comment]
Nice. I have a few suggestions that I think would make this PR even better:

- Make the two lists returned by `plot_by_id` have the same length with `None` in the colorbar axis list where no colorbar is present.
- Change `cax` to `cbax` and `caxes` to `cbaxes`.
- Update the `Testing quick plotting` notebook. The name is poor and could be changed to `Quick Plot Tutorial` or something like that. The notebook should briefly explain what `plot_by_id` does and show one or two fancy examples of using many subplots for arrangement.

I can do some of the above, if you wish.

**Comment by jenshnielsen** [comment]
I agree, Feel free to do the changes which will also give you a chance to test it out

**Comment by jenshnielsen** [comment]
@WilliamHPNielsen I addressed the first 2 issues. Do you want to update the notebook

**Comment by WilliamHPNielsen** [comment]
> Do you want to update the notebook?

Yes, I am on it now.

**Comment by jenshnielsen** [comment]
Just waiting for travis following the merge of the mypy branch

File: qcodes/dataset/plotting.py [hunk 1]
```diff
@@ -142,6 +142,10 @@ def set_axis_labels(ax, data, cax=None):
                        f'{len(data-1)} parameters, cannot plot '
                        f'that.')
             cbaxes.append(None)
+
+    if len(axes) != len(cbaxes):
+        raise RuntimeError("Non equal number of axes. Perhaps cbaxes is missing"
+                           "from one of the cases above")
```
Comment (WilliamHPNielsen) [inline qcodes/dataset/plotting.py]: Detail: missing space between "missing" and "from".

Comment (jenshnielsen) [reply qcodes/dataset/plotting.py]: #1031

**Pull request merged**